src/spac/visualization.py

```diff
@@ -251,22 +274,47 @@ def dimensionality_reduction_plot(
             )
         key = associated_table
 
-    logger = logging.getLogger()
-    logger.info(f'Running visualizaiton using the coordinates:"{key}"')
+    print(f'Running visualization using the coordinates: "{key}"')
 
     # Extract the 2D coordinates
     x, y = adata.obsm[key].T
 
     # Determine coloring scheme
     if annotation:
         color_values = adata.obs[annotation].astype('category').values
+        color_representation = annotation
     elif feature:
         data_source = adata.layers[layer] if layer else adata.X
         color_values = data_source[:, adata.var_names == feature].squeeze()
+        color_representation = feature
     else:
         color_values = None
-
-    fig, ax = visualize_2D_scatter(x, y, ax=ax, labels=color_values, **kwargs)
+        color_representation = None
+
+    # Set axis titles based on method
+    if method == 'tsne':
+        x_axis_title = 't-SNE 1'
+        y_axis_title = 't-SNE 2'
+        plot_title = 'TSNE'
+    elif method == 'pca':
+        x_axis_title = 'PCA 1'
+        y_axis_title = 'PCA 2'
+        plot_title = 'PCA'
+    elif method == 'umap':
+        x_axis_title = 'UMAP 1'
+        y_axis_title = 'UMAP 2'
+        plot_title = 'UMAP'
+    else:

```

Comment:
Hi @fangliu117, for this condition, the x and y title might not be UMAP, would something like f"{associated_table} 1" and 2 make more sense? Thanks.

Reply:
or just leave it empty since default is component 1 and component 2

Reply:
Thank you @ruiheesi for your feedback. You are correct that assuming the x and y axis titles to be UMAP 1 and UMAP 2 are not appropriate when an associated_table is provided. To ensure the axis titles are contextually accurate, I have updated the function to dynamically generate the axis titles based on the associated_table name.